lib/MetaCPAN/Model/Search.pm

```diff
@@ -122,32 +138,45 @@ sub _search_collapsed {
             fields => [qw(distribution)],
         };
 
-        # On the first request also fetch the number of total distributions
-        # that match the query so that can be reported to the user. There is
-        # no need to do it on each iteration though, once is enough.
-        $es_query_opts->{aggregations}
-            = {
-            count => { terms => { size => 999, field => 'distribution' } }
-            }
-            if $run == 1;
-        my $es_query = $self->build_query( $query, $es_query_opts );
-
-        $data = $self->run_query( file => $es_query );
-        $took += $data->{took} || 0;
-        $total = @{ $data->{aggregations}->{count}->{buckets} || [] }
-            if $run == 1;
-        $hits = @{ $data->{hits}->{hits} || [] };
+        if ( $run == 1 ) {
+
+          # On the first request also fetch the number of total distributions
+          # that match the query so that can be reported to the user. There is
+          # no need to do it on each iteration though, once is enough.
+
+            $es_query_opts->{aggregations}
+                = {
+                count => { terms => { size => 999, field => 'distribution' } }
+                };
+
+        }
+
+        # Build the query and fetch results
+        my $es_query = $self->build_query( $search_term, $es_query_opts );
+        $es_results = $self->run_query( file => $es_query );
+
+        if ( $run == 1 ) {
```

Comment:
basically so it jumped up and slapped you in the face, rather than being a `..... if $run == 1;`